lighthouse-core/test/lib/dependency-graph/simulator/simulator-test.js

```diff
@@ -118,6 +118,24 @@ describe('DependencyGraph/Simulator', () => {
       assertNodeTiming(result, nodeB, {startTime: 8, endTime: 16});
     });
 
+    it('should simulate data URL network graphs', () => {
+      const url = 'data:image/jpeg;base64,foobar';
+      const protocol = 'data';
+      const parsedURL = {scheme: 'data', host: '', securityOrigin: 'null'};
+      const nodeA = new NetworkNode(request({startTime: 0, endTime: 1, url, parsedURL, protocol}));
+      const nodeB = new NetworkNode(request({startTime: 0, endTime: 3, url, parsedURL, protocol,
+        resourceSize: 1024 * 1024}));
+      nodeA.addDependent(nodeB);
+
+      const simulator = new Simulator({serverResponseTimeByOrigin});
+      const result = simulator.simulate(nodeA);
+
+      // should be ~2ms for A, ~12ms for B
```

Comment:
for clarity for future tinkerers, maybe
```suggestion
      // should be ~2ms for A (resourceSize 0), ~12ms for B (resourceSize 1MB)
```
...or else a comment up on the node creation lines calling out the resourceSize?